pkg/primaza/workercluster/agentapp.go

```diff
@@ -0,0 +1,225 @@
+/*
+Copyright 2023 The Primaza Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package workercluster
+
+import (
+	"context"
+	"fmt"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/client-go/kubernetes"
+)
+
+func DeleteApplicationAgent(ctx context.Context, cli *kubernetes.Clientset, namespace string) error {
+	if err := deleteAgentAppDeployment(ctx, cli, namespace); err != nil {
+		return err
+	}
+
+	if err := deleteAgentAppService(ctx, cli, namespace); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func PushApplicationAgent(ctx context.Context, cli *kubernetes.Clientset, namespace string) error {
+	if _, err := createAgentAppDeployment(ctx, cli, namespace); err != nil && !errors.IsAlreadyExists(err) {
+		return err
+	}
+
+	if _, err := createAgentAppService(ctx, cli, namespace); err != nil && !errors.IsAlreadyExists(err) {
```

Comment:
why do we need a service for the agents?

Reply:
you're right it's not needed, I added this debugging the problems related to leases and forgot it here. nice catch!